src/IO/S3/copyS3File.h

```diff
@@ -17,21 +17,18 @@ class SeekableReadBuffer;
 
 using CreateReadBuffer = std::function<std::unique_ptr<SeekableReadBuffer>()>;
 
-/// Copies a file from S3 to S3.
+/// Copies a file from S3 to S3 using native copy (CopyObject).
 /// The same functionality can be done by using the function copyData() and the classes ReadBufferFromS3 and WriteBufferFromS3
 /// however copyS3File() is faster and spends less network traffic and memory.
 /// The parameters `src_offset` and `src_size` specify a part in the source to copy.
 ///
-/// Note, that it tries to copy file using native copy (CopyObject), but if it
-/// has been disabled (with settings.allow_native_copy) or request failed
-/// because it is a known issue, it is fallbacks to read-write copy
-/// (copyDataToS3File()).
-///
 /// s3_client_with_long_timeout (may be equal to s3_client) is used for native copy and
 /// CompleteMultipartUpload requests. These requests need longer timeout because S3 servers often
 /// block on them for multiple seconds without sending or receiving data from us (maybe the servers
 /// are copying data internally, or maybe throttling, idk).
-void copyS3File(
+///
+/// @return false if native copy is not supported
```

Comment:
This interface is not very clear. Usually a function returns false when it fails (as an alternative to throwing an exception).

Reply:
The API that fallbacks to the regular copy does not sounds clean to me as well, but this is another story.